samples/opengl/opengl3_2.cpp

```diff
@@ -0,0 +1,169 @@
+#include <iostream>
+
+#include <epoxy/gl.h>
+
+#ifdef _WIN32
+    #define WIN32_LEAN_AND_MEAN 1
+    #define NOMINMAX 1
+    #include <windows.h>
+#endif
+
+#if defined(__APPLE__)
+    #include <OpenGL/gl.h>
+    #include <OpenGL/glu.h>
+#else
+    #include <GL/gl.h>
+    #include <GL/glu.h>
+#endif
+
+#include "opencv2/core.hpp"
+#include "opencv2/core/opengl.hpp"
+#include "opencv2/core/cuda.hpp"
+#include "opencv2/highgui.hpp"
+
+using namespace std;
+using namespace cv;
+using namespace cv::cuda;
+
+const int win_width = 800;
+const int win_height = 640;
+
+struct DrawData
+{
+    GLuint vao, vbo, program, textureID;
+};
+
+cv::Mat rot(float angle)
+{
+    cv::Mat R_y = (cv::Mat_<float>(4,4) <<
+        cos(angle), 0, sin(angle), 0,
+        0, 1, 0, 0,
+        -sin(angle), 0, cos(angle), 0,
+        0, 0, 0, 1);
+
+    return R_y;
+}
+
+static GLuint create_shader(const char* source, GLenum type) {
+    GLuint shader = glCreateShader(type);
+    glShaderSource(shader, 1, &source, NULL);
+    glCompileShader(shader);
+    return shader;
+}
+
+void draw(void* userdata){
+    DrawData* data = static_cast<DrawData*>(userdata);
+    static float angle = 0.0f;
+    angle += 1.f;
+
+    cv::Mat trans = rot(CV_PI * angle / 360.f);
+
+    glClearColor(0.0, 0.0, 0.0, 1.0);
+    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+
+    glUseProgram(data->program);
+    glUniformMatrix4fv(glGetUniformLocation(data->program, "transform"), 1, GL_FALSE, trans.ptr<float>());
+    glBindTexture(GL_TEXTURE_2D, data->textureID);
+    glBindVertexArray(data->vao);
+    glDrawArrays(GL_TRIANGLE_STRIP, 0, 4);
+    glBindVertexArray(0);
+}
+
+int main(int argc, char* argv[])
+{
+    string filename;
+    if (argc < 2)
+    {
+        cout << "Usage: " << argv[0] << " image" << endl;
+        filename = "lena.jpg";
```

Comment:
similar to #25635 - perhaps choose a different default file name from the https://github.com/opencv/opencv/tree/4.x/samples/data samples e.g.
```suggestion
        filename = "HappyFish.jpg";
```

Reply:
https://github.com/opencv/opencv/pull/25822/commits/174265724cb21160229b6b5591acba3526fbc261